### PR TITLE
remove-w2ui #13: mobile bottom bar (final patch)

### PR DESCRIPTION
**Pull request by eszkadev**
- converted mobile bottom bar
- removed w2ui JS file (CSS left, need to investigate used rules, eg. we need to rescue w2ui-right-scroll used for scrolling NB or toolbars)
- toolitems got select/unselect action
- mobile uses desktop's snackbar (old implementation can be later removed from mobile-wizard)
- mobile can show dropdowns using desktop's JSDialog component
- moved JSDialog message routing to separate file from Socket

### CODE REVIEW

File: browser/src/control/Control.JSDialogBuilder.js [hunk 1]
```diff
@@ -2424,10 +2424,21 @@
...
-				$(div).addClass('selected');
+			var selectFn = function() {
+				L.DomUtil.addClass(button, 'selected');
+				L.DomUtil.addClass(div, 'selected');
 			}
+
+			var unSelectFn = function() {
+				L.DomUtil.removeClass(button, 'selected');
+				L.DomUtil.removeClass(div, 'selected');
+			}
```
[the remark this answers is not in the document]
Comment (eszkadev) [reply browser/src/control/Control.JSDialogBuilder.js]: done

File: browser/src/docdispatcher.ts [hunk 2]
```diff
@@ -529,6 +550,16 @@ class Dispatcher {
 				data: (window as any).getColorPickerData('Highlight Color'),
 			});
 		};
+		// TODO: leftover from mobile bottom bar
```
Comment (eszkadev) [inline browser/src/docdispatcher.ts:553]: I will later prepare cleanup patch, ignore todos and comments :)

**Comment by eszkadev** [comment]
> Just a general comment for this and future js lib removals: please update THIRDPARTYLICENSES file, when you remove something.

Ok, we still have part of that library (CSS), after complete removal I will update THIRDPARTYLICENSES

**Review by gokaysatir — APPROVED** [review]
Thanks @eszkadev :) This is a big step forward :)

There is a duplicate item in the bottom bar of Writer. And also i guess we can consult @pedropintosilva if we want an status bar in mobile view. I couldn't see it. We can consider these in a follow up if you wish.

**Comment by eszkadev** [comment]
> Thanks @eszkadev :) This is a big step forward :)
> 
> There is a duplicate item in the bottom bar of Writer. And also i guess we can consult @pedropintosilva if we want an status bar in mobile view. I couldn't see it. We can consider these in a follow up if you wish.
> 
> 

Thank you for testing, I will fix it in the followup cleanup

**Pull request merged**